### ISSUE TITLE
Error when running asv on local repo without master branch

### ISSUE THREAD

**Issue by bsipocz** [airspeed-velocity/asv#690]
I'm trying to use asv locally to check on possible performance improvements on a branch I work on. I don't have a `'master'` branch though and thus run into this error:

```
$ asv run 47d6879f93dbed0203f6bd879a60132f13a004fe..3e7e5783a03de2d22f8aff397e531db2e152dd58 --bench time_sort
· Creating environments..................................
· Discovering benchmarks
·· Error running /opt/local/bin/git rev-parse master
             STDOUT -------->
             master
             STDERR -------->
             fatal: ambiguous argument 'master': unknown revision or path not in the working tree.
             Use '--' to separate paths from revisions, like this:
             'git <command> [<revision>...] -- [<file>...]'
Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/bin/asv", line 11, in <module>
    load_entry_point('asv==0.2.2', 'console_scripts', 'asv')()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/asv/main.py", line 38, in main
    result = args.func(args)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/asv/commands/__init__.py", line 48, in run_from_args
    return cls.run_from_conf_args(conf, args)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/asv/commands/run.py", line 127, in run_from_conf_args
    **kwargs
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/asv/commands/run.py", line 185, in run
    benchmarks = Benchmarks(conf, repo, environments, regex=bench)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/asv/benchmarks.py", line 299, in __init__
    benchmarks = self.disc_benchmarks(conf, repo, environments)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/asv/benchmarks.py", line 340, in disc_benchmarks
    env.install_project(conf, repo)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/asv/environment.py", line 514, in install_project
    commit_hash = repo.get_hash_from_name(conf.branches[0])
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/asv/plugins/git.py", line 131, in get_hash_from_name
    dots=False).strip().split()[0]
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/asv/plugins/git.py", line 73, in _run_git
    return util.check_output([self._git] + args, env=env, **kwargs)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/asv/util.py", line 526, in check_output
    raise ProcessError(args, retcode, stdout, stderr)
asv.util.ProcessError: Command '/opt/local/bin/git rev-parse master' returned non-zero exit status 128
```

The current workaround is to create a `'master'` branch, though it would be nice not being forced to do this.

**Comment by pv** [comment]
This has been fixed in the development version, #674

**Comment by bsipocz** [comment]
Oh brilliant thanks! 
I'm sorry for the noise of not checking the dev version first (and only looking for open issues not the closed ones).
